pkg/slo-controller/noderesource/plugins/gpudeviceresource/plugin.go

```diff
@@ -149,26 +151,39 @@ func (p *Plugin) Calculate(_ *configuration.ColocationStrategy, node *corev1.Nod
 		return p.resetGPUNodeResource()
 	}
 
+	useDPGPUSharedResource := false
+	if utilfeature.DefaultFeatureGate.Enabled(features.EnableSyncGPUSharedResource) {
+		useDPGPUSharedResource = true
+	}
+
 	// TODO: calculate NUMA-level resources against NRT
-	return p.calculate(node, device)
+	return p.calculate(useDPGPUSharedResource, node, device)
 }
 
-func (p *Plugin) calculate(node *corev1.Node, device *schedulingv1alpha1.Device) ([]framework.ResourceItem, error) {
+func (p *Plugin) calculate(useDPGPUSharedResource bool, node *corev1.Node, device *schedulingv1alpha1.Device) ([]framework.ResourceItem, error) {
 	if device == nil {
 		return nil, fmt.Errorf("invalid device")
 	}
 
 	// calculate gpu resources
 	gpuResources := make(corev1.ResourceList)
 	totalKoordGPU := resource.NewQuantity(0, resource.DecimalSI)
+	healthGPUNum := 0
 	for _, d := range device.Spec.Devices {
 		if d.Type != schedulingv1alpha1.GPU || !d.Health {
 			continue
 		}
+
+		healthGPUNum++
 		util.AddResourceList(gpuResources, d.Resources)
 		totalKoordGPU.Add(d.Resources[extension.ResourceGPUCore])
 	}
+
 	gpuResources[extension.ResourceGPU] = *totalKoordGPU
+	if !useDPGPUSharedResource {
```

Comment:
After adjusting the semantics, should the conditional judgment here be reversed?

if enableSyncGPUSharedResource ?

Reply:
sorry，I miss this place

Reply:
Once you have reviewed everything, I will make all the necessary modifications at once

Reply:
fixed